datadog/dogstatsd/base.py

```diff
@@ -598,7 +642,35 @@ def disable_buffering(self, is_disabled):
                 log.debug("Statsd buffering is disabled")
             else:
                 self._send = self._send_to_buffer
-                self._start_flush_thread(self._flush_interval)
+                self._start_flush_thread(
+                    self._flush_interval,
+                    self.flush_buffered_metrics,
+                )
+
+    def disable_aggregation(self):
+        with self._config_lock:
+            # If the toggle didn't change anything, this method is a noop
+            if self._disable_aggregating:
+                return
+
+            self._disable_aggregating = True
+
+            # If aggregation has been disabled, flush and kill the background thread
+            # otherwise start up the flushing thread and enable aggregation.
+            self._stop_flush_thread()
+            log.debug("Statsd aggregation is disabled")
+
+    def enable_aggregation(self, aggregation_flush_interval=DEFAULT_AGGREGATION_FLUSH_INTERVAL):
```

Comment:
the __init__ function is sometimes not called with any parameters set and just has the default parameters as we don't require the user to call initialize before sending metrics. So I believe we want a way to start aggregation.

This was a convo I had with Vikentiy 

"That’s because we create the global statsd instance with all default parameters ([here](https://github.com/DataDog/datadogpy/blob/add-client-side-aggregation/datadog/dogstatsd/base.py#L1630)) - we do not require user to call initialize before sending metrics. Then, if the user calls configuration routine, we change the state of the object as needed. We would need at least a way to enable aggregation on an object that already exists, ..."

Reply:
Oh awesome. Ill give this another look 👍